doc: Update 'Netplan everywhere' for 23.10 (#418)
canonical/netplan: doc/netplan-everywhere.md

# Desktop integration

## NetworkManager YAML settings backend

NetworkManager is the tool used by Ubuntu Desktop systems to manage
network devices such as Ethernet and Wifi adapters. While it is a great
tool for the job and users can directly use it through the command line
and the graphical interfaces to configure their devices, Ubuntu has its
own way of describing and storing network configuration via Netplan.

On Ubuntu 23.10 "Mantic Minotaur" and later, NetworkManager uses Netplan's APIs
to save the configuration created through any of its graphical or programmatic
interfaces. This leads to having a centralized location to store network
configuration. In the Desktop it's convenient to use graphical tools for
configuration when they are available, so nothing will change from the user

> perspective, only the way the configuration is handled under the hood.

Reviewer: ```suggestion
perspective; only the way the system handles the configuration in the background.
```

> For more information on Netplan, check [https://netplan.io](https://netplan.io)

Reviewer: ```suggestion
For more information on Netplan, see [https://netplan.io](https://netplan.io).
```

> For more information on NetworkManager, check [https://networkmanager.dev](https://networkmanager.dev)

Reviewer: ```suggestion
For more information on NetworkManager, see [https://networkmanager.dev](https://networkmanager.dev).
```

## How it works

Every time a non-temporary connection is created in NetworkManager, instead

> of persisting the original `.nmconnection` file, it will create a Netplan YAML

Reviewer: ```suggestion
of persisting the original `.nmconnection` file, it creates a Netplan YAML
```
